### PR TITLE
Mochawesome reporting

### PR DESCRIPTION
**Pull request by thekiiingbob**
#### Summary
Use the cypress module to configure cypress and run tests programatically
Integrate mochawesome reports (pretty HTML reports, with integrated screenshots)

To run, just do `npm run test`, afterwards you should have a mochawesome-report directory. If you open up mochawesome.html in your browser, you can view the report.

#### Ticket Link
[JIRA](https://mattermost.atlassian.net/browse/MM-16834)

### CODE REVIEW

File: e2e/runTests.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        });
+
+        totalFailed += failed;
+    }
+
+    const jsonReport = await merge(); // generate JSON report
+
+    await generator.create(jsonReport);
+
+    await fse.copy('screenshots', 'mochawesome-report/screenshots');
```
Comment (saturninoabril) [inline e2e/runTests.js]: 0/5, could we expose a single top-level folder for all those auto-generated files?  Maybe all in `/e2e/reports`; folder structure could be:
```
|-- e2e
    |-- report
        |-- mochawesome-report
        |-- results
        |-- screenshots
```

One thing that it solves is to easily ignore-pattern for ESLint rules (`--ignore-pattern e2e/report`), as it's complaining too much when you ran ESLint with files under `e2e/mochawesome-report`, specifically the `app.js` on it.

Comment (thekiiingbob) [reply e2e/runTests.js]: Will put everything in results/ directory. Will update the pipeline PR as well after I verify some of the other kinks.

**Comment by thekiiingbob** [comment]
> This is great!
> 
> Could you please check once again the generated link for screenshot? It's broken on me when I ran it. There were `/` that's missing in the link.
> Screen Shot 2019-07-16 at 2 54 02 PM

Ah! Good catch! I was stripping out `/` from the final filename, I just needed to strip it from the parent names.

**Comment by thekiiingbob** [comment]
@saturninoabril @prapti Everything should be up to date now!

**Comment by prapti** [comment]
Should I be looking for the mochawesome-report folder inside e2e/results?
I am however, getting the following error while trying to run the test. Am I missing anything?

> Error: Command failed: /bin/sh -c npm run test:ci
>     at makeError (/Users/prapti/go/src/github.com/mattermost/mattermost-webapp/e2e/node_modules/start-server-and-test/node_modules/execa/index.js:174:9)
>     at /Users/prapti/go/src/github.com/mattermost/mattermost-webapp/e2e/node_modules/start-server-and-test/node_modules/execa/index.js:278:16
>     at processTicksAndRejections (internal/process/task_queues.js:85:5) {
>   code: 1,
>   stdout: null,
>   stderr: null,
>   failed: true,
>   signal: null,
>   cmd: '/bin/sh -c npm run test:ci',
>   timedOut: false,
>   killed: false
> }
> npm ERR! code ELIFECYCLE
> npm ERR! errno 1
> npm ERR! @ test: `start-test start:webhook 3000 test:ci`
> npm ERR! Exit status 1
> npm ERR! 
> npm ERR! Failed at the @ test script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

File: .gitignore [hunk 2]
```diff
@@ -12,6 +12,8 @@ mattermost-webapp.iml
 
 # disable folders generated by Cypress
 e2e/node_modules
-e2e/cypress/screenshots
+e2e/screenshots
 e2e/cypress/videos
 e2e/results
+e2e/mochawesome-report
+e2e/mochawesome-merged.json
```
Comment (saturninoabril) [inline .gitignore]: Let's clean this up with the only folders to ignore.

Comment (thekiiingbob) [reply .gitignore]: @saturninoabril Pushed up changes to remove those unneeded ignores. Added `.eslintignore` file to handle ignoring e2e/results in linting.

**Pull request merged**